pkg/providers/nutanix/validator.go

```diff
@@ -0,0 +1,195 @@
+package nutanix
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	prismgoclient "github.com/nutanix-cloud-native/prism-go-client"
+	"github.com/nutanix-cloud-native/prism-go-client/utils"
+	"github.com/nutanix-cloud-native/prism-go-client/v3"
+	"go.uber.org/multierr"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+// Validator is a client to validate nutanix resources
+type Validator struct {
+	v3Client client
+}
+
+// NewValidator returns a new validator client
+func NewValidator(config *anywherev1.NutanixDatacenterConfig, creds basicAuthCreds) (*Validator, error) {
+	url := fmt.Sprintf("%s:%d", config.Spec.Endpoint, config.Spec.Port)
+	nutanixCreds := prismgoclient.Credentials{
+		URL:      url,
+		Username: creds.username,
+		Password: creds.password,
+		Endpoint: config.Spec.Endpoint,
+	}
+	client, err := v3.NewV3Client(nutanixCreds)
```

Comment:
Can we inject the client instead of constructing it in-line. In doing so, we create a more testable data structure and avoid twiddling internals in tests as we have to now.